docs/design/phases.md

```diff
@@ -0,0 +1,110 @@
+## Phases in kubeadm
+
+`kubeadm init` does a lot of things. In order to make kubeadm the common building block for Kubernetes we want it to be, each logical phase kubeadm does during `init` should be invokable separately. Here's an explanation what kubeadm does in the different phases:
+
+#### A note on constants / well-known values and paths
+
+Somewhere we have to draw the line what should be configurable, what shouldn't and what should be hard-coded in the binary.
```

Comment:
Suggested edit:

We have to draw the line somewhere about what should be configurable, what shouldn't, and what should be hard-coded in the binary.